File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    int letters, words, sentences;

    string text = get_string("Text: ");

    letters = sentences = 0;
    words = 1;

    for (int a = 0; a < strlen(text); a++)
    {
        if (isalpha(text[a]))
        {
            letters++;
        }
        else if (text[a] == ' ')
        {
            words++;
        }

        else if (text[a] == '.' || text[a] == '?' || text[a] == '!')
        {
            sentences++;
        }
    }

    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

string = get_string("Text: ").strip()
num_words = num_letters = num_setences = 0

for i in range(len(string)):
    if (i == 0 and string[i] != ' ') or (i != len(string) - 1 and string[i] == ' ' and string[i + 1] != ' '):
        num_words += 1
    if string[i].isalpha():
        num_letters += 1
    if string[i] == '.' or string[i] == '?' or string[i] == '!':
        num_setences += 1

L = num_letters / num_words * 100
S = num_setences / num_words * 100
index = round(0.0588 * L - 0.296 * S - 15.8)
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")